gen.go

```diff
@@ -259,6 +262,8 @@ func tagparse(v string) (map[string]string, error) {
 			out[strings.TrimSpace(parts[0])] = strings.TrimSpace(parts[1])
 		} else if elem == "omitempty" {
 			out["omitempty"] = "true"
+		} else if elem == "preservenil" {
+			out["preservenil"] = "true"
```

Comment:
uber nit: We may want to fail gen if this struct tag is applied to something other than a slice, since it's ignored everywhere else (as opposed to all other struct tags, which can be applied to any field).